### PR TITLE
Add alias docs(man pages) for nmstatectl subcommands

### LINKED ISSUE #2579: Create alias for nmstaetctl subcommand
Considering these two subcommand are widely used by developer and user, to save some typing, we can create alias like:

 * `nmstatectl a` == `nmstatectl apply`
 * `nmstatectl s` == `nmstatectl show`

This is not about bash autocomplet, but rust clap alias.

### PR DESCRIPTION
**Pull request by jona42-ui**
Added aliases docs for  a and  s, providing shortcuts for apply and
show respectively for simplified command invocation.

Resolves: https://github.com/nmstate/nmstate/issues/2579

### CODE REVIEW

File: rust/src/cli/ncl.rs [hunk 1]
```diff
@@ -138,7 +139,7 @@ fn main() {
         .subcommand(
             clap::Command::new(SUB_CMD_APPLY)
                 .about("Apply network state or network policy")
-                .alias("set")
```
Comment (ihuguet) [inline rust/src/cli/ncl.rs]: Please do not remove existing aliases, people might be using them and they will suddenly stop working.

Comment (jona42-ui) [reply rust/src/cli/ncl.rs]: thanks so much @ihuguet reverting that

Comment (jona42-ui) [reply rust/src/cli/ncl.rs]: any pointer about the failed checks @ihuguet 
one seem to be related to fedora

Comment (jona42-ui) [reply rust/src/cli/ncl.rs]: Also I suppose we need to update the documents for users to know of the new aliases

Comment (ihuguet) [reply rust/src/cli/ncl.rs]: I think you can ignore the Fedora failure, it seems a dependency issue to a package that is not updated in Fedora yet. Certainly not related to your changes.

About the documentation: good point! Honestly I don't know if we have any of the other alias documented.

**Review by ihuguet — APPROVED** [review]

**Comment by ihuguet** [comment]
Approved. Just a nit: your sign-off contains your alias instead of your name. If it's not a problem to put your name, I think it's prefered because in Open Source working with "real" people is better.

**Comment by jona42-ui** [comment]
> Approved. Just a nit: your sign-off contains your alias instead of your name. If it's not a problem to put your name, I think it's prefered because in Open Source working with "real" people is better.

thank you @ihuguet , I can see the aliases affecting me. it seems github picks my username not my real name(not sure how it works with the -s flag) but I will change. thanks

**Comment by jona42-ui** [comment]
@cathay4t  does this meet the merge criterion?

**Review by cathay4t — CHANGES_REQUESTED** [review]
Please mention this alias in manpage `doc/nmstatectl.8.in`

**Comment by jona42-ui** [comment]
okay thanks

**Comment by jona42-ui** [comment]
> Please mention this alias in manpage `doc/nmstatectl.8.in`

I have never edited this type of file, I hope all goes well

**Comment by jona42-ui** [comment]
cant explicitly tell why the integration test is failing

**Comment by liangwen12year** [comment]
The following is the result after running `man doc/nmstatectl.8.in`, IMHO, the important thing to highlight should be that `nmstatectl s` is an alias or equivalent to  `nmstatectl show` and  `nmstatectl a` is an alias or equivalent to  `nmstatectl apply`. Otherwise, it will only confuse the user and user might think that `nmstatectl s` is  a command for different purpose compared to `nmstatectl show`. 


```
nmstatectl(8)                                           nmstatectl man page                                          nmstatectl(8)

NAME
       nmstatectl - A nmstate command line tool

SYNOPSIS
       nmstatectl show [INTERFACE_NAME] [--json]
       nmstatectl show [-r, --running-config]
       nmstatectl s [INTERFACE_NAME] [--json]
       nmstatectl s [-r, --running-config]
       nmstatectl set STATE_FILE_PATH [OPTIONS]
       nmstatectl apply STATE_FILE_PATH [OPTIONS]
       nmstatectl a STATE_FILE_PATH [OPTIONS]
       nmstatectl edit [INTERFACE_NAME] [OPTIONS]
       nmstatectl gc [STATE_FILE_PATH] [OPTIONS]
       nmstatectl rollback [CHECKPOINT_PATH]
       nmstatectl commit [CHECKPOINT_PATH]
       nmstatectl service [-c, --config <CONFIG_FOLDER>]
       nmstatectl statistic STATE_FILE_PATH [-c, --current <CURRENT_STATE_FILE>]
       nmstatectl version

DESCRIPTION
       nmstatectl is created for users who want to try out nmstate without using libnmstate.
```

Thus, I think that we should not mention in the `SYNOPSIS` but in the `DESCRIPTION` section:

```
       show
              Query  the  current  network  state. YAML is the default output format. Use the --json argument to change the output
              format to JSON. To limit the output state to include certain interfaces only, please  specify  the  interface  name.
              Please be advised, global config like DNS will be included. `nmstatectl s` is an aliased command for `nmstatectl show`. 
```
@ihuguet @cathay4t , what do you think ?

**Comment by liangwen12year** [comment]
@jona42-ui , Also, the commit message is not properly formatted, https://github.com/nmstate/nmstate/blob/base/CONTRIBUTING.md#write-a-good-commit-message, please follow the instructions in the link to wrap the body at 72 characters.

**Comment by jona42-ui** [comment]
> The following is the result after running `man doc/nmstatectl.8.in`, IMHO, the important thing to highlight should be that `nmstatectl s` is an alias or equivalent to `nmstatectl show` and `nmstatectl a` is an alias or equivalent to `nmstatectl apply`. Otherwise, it will only confuse the user and user might think that `nmstatectl s` is a command for different purpose compared to `nmstatectl show`.
> 
> ```
> nmstatectl(8)                                           nmstatectl man page                                          nmstatectl(8)
> 
> NAME
>        nmstatectl - A nmstate command line tool
> 
> SYNOPSIS
>        nmstatectl show [INTERFACE_NAME] [--json]
>        nmstatectl show [-r, --running-config]
>        nmstatectl s [INTERFACE_NAME] [--json]
>        nmstatectl s [-r, --running-config]
>        nmstatectl set STATE_FILE_PATH [OPTIONS]
>        nmstatectl apply STATE_FILE_PATH [OPTIONS]
>        nmstatectl a STATE_FILE_PATH [OPTIONS]
>        nmstatectl edit [INTERFACE_NAME] [OPTIONS]
>        nmstatectl gc [STATE_FILE_PATH] [OPTIONS]
>        nmstatectl rollback [CHECKPOINT_PATH]
>        nmstatectl commit [CHECKPOINT_PATH]
>        nmstatectl service [-c, --config <CONFIG_FOLDER>]
>        nmstatectl statistic STATE_FILE_PATH [-c, --current <CURRENT_STATE_FILE>]
>        nmstatectl version
> 
> DESCRIPTION
>        nmstatectl is created for users who want to try out nmstate without using libnmstate.
> ```
> 
> Thus, I think that we should not mention in the `SYNOPSIS` but in the `DESCRIPTION` section:
> 
> ```
>        show
>               Query  the  current  network  state. YAML is the default output format. Use the --json argument to change the output
>               format to JSON. To limit the output state to include certain interfaces only, please  specify  the  interface  name.
>               Please be advised, global config like DNS will be included. `nmstatectl s` is an aliased command for `nmstatectl show`. 
> ```
> 
> @ihuguet @cathay4t , what do you think ?

thanks so much @liangwen12year for the pointers.
I would add that adding them to the description is a good insight.
But I suggest we keep them(aliases) both at synopsis and description.
First to let the user know they exist, then let them know they are  aliases in the description.

**Comment by jona42-ui** [comment]
part of this was commited here: https://github.com/nmstate/nmstate/pull/2633

so let me deal with docs

**Comment by liangwen12year** [comment]
> > The following is the result after running `man doc/nmstatectl.8.in`, IMHO, the important thing to highlight should be that `nmstatectl s` is an alias or equivalent to `nmstatectl show` and `nmstatectl a` is an alias or equivalent to `nmstatectl apply`. Otherwise, it will only confuse the user and user might think that `nmstatectl s` is a command for different purpose compared to `nmstatectl show`.
> > ```
> > nmstatectl(8)                                           nmstatectl man page                                          nmstatectl(8)
> > 
> > NAME
> >        nmstatectl - A nmstate command line tool
> > 
> > SYNOPSIS
> >        nmstatectl show [INTERFACE_NAME] [--json]
> >        nmstatectl show [-r, --running-config]
> >        nmstatectl s [INTERFACE_NAME] [--json]
> >        nmstatectl s [-r, --running-config]
> >        nmstatectl set STATE_FILE_PATH [OPTIONS]
> >        nmstatectl apply STATE_FILE_PATH [OPTIONS]
> >        nmstatectl a STATE_FILE_PATH [OPTIONS]
> >        nmstatectl edit [INTERFACE_NAME] [OPTIONS]
> >        nmstatectl gc [STATE_FILE_PATH] [OPTIONS]
> >        nmstatectl rollback [CHECKPOINT_PATH]
> >        nmstatectl commit [CHECKPOINT_PATH]
> >        nmstatectl service [-c, --config <CONFIG_FOLDER>]
> >        nmstatectl statistic STATE_FILE_PATH [-c, --current <CURRENT_STATE_FILE>]
> >        nmstatectl version
> > 
> > DESCRIPTION
> >        nmstatectl is created for users who want to try out nmstate without using libnmstate.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Thus, I think that we should not mention in the `SYNOPSIS` but in the `DESCRIPTION` section:
> > ```
> >        show
> >               Query  the  current  network  state. YAML is the default output format. Use the --json argument to change the output
> >               format to JSON. To limit the output state to include certain interfaces only, please  specify  the  interface  name.
> >               Please be advised, global config like DNS will be included. `nmstatectl s` is an aliased command for `nmstatectl show`. 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @ihuguet @cathay4t , what do you think ?
> 
> thanks so much @liangwen12year for the pointers. I would add that adding them to the description is a good insight. But I suggest we keep them(aliases) both at synopsis and description. First to let the user know they exist, then let them know they are aliases in the description.

I am fine if you want to also keep it in synopsis, but you need to figure out a way to make sure that the user will think that `nmstatectl s` is an alias or equivalent to `nmstatectl show` and the user is not confused about the usage of `nmstatectl s`

File: doc/nmstatectl.8.in [hunk 2]
```diff
@@ -7,10 +7,16 @@ nmstatectl \- A nmstate command line tool
 .br
 .B nmstatectl show [\fB-r, --running-config\fR]
 .br
+.B nmstatectl s \fR[\fIINTERFACE_NAME\fR] [\fB--json\fR]
```
Comment (cathay4t) [inline doc/nmstatectl.8.in]: This give user impression that `nmstatectl s` is a new subcommand, and every time we have new option added to `nmtatectl show`, we have to do it for this `nmstatectl s`.

I prefer not listing it at top level, just mention in description of `show` that you can use `s` as alias to save typing.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: doc/nmstatectl.8.in (hunk 2 above)
Comment (jona42-ui) [reply doc/nmstatectl.8.in]: > This give user impression that `nmstatectl s` is a new subcommand, and every time we have new option added to `nmtatectl show`, we have to do it for this `nmstatectl s`.
> 
> I prefer not listing it at top level, just mention in description of `show` that you can use `s` as alias to save typing.

thanks. updated, but ci is not happy yet

File: doc/nmstatectl.8.in [hunk 3]
```diff
@@ -65,6 +66,7 @@ Apply the network state from specified file in \fIYAML\fR or \fIJSON\fR format.
 By default, if the network state after state applied is not identical to the
 desired state, \fBnmstatectl\fR rollbacks to the state before \fBset\fR
 command. Use the \fB--no-verify\fR argument to skip the verification.
+\fBnmstatectl a\fR is an aliased command for \fBnmstatectl apply\fR.
```
Comment (liangwen12year) [inline doc/nmstatectl.8.in]: This should be moved to the `apply` section below.

Comment (jona42-ui) [reply doc/nmstatectl.8.in]: > This should be moved to the `apply` section below.

oops! my bad

Comment (jona42-ui) [reply doc/nmstatectl.8.in]: updated!

File: doc/nmstatectl.8.in (hunk 2 above)
Comment (jona42-ui) [reply doc/nmstatectl.8.in]: happy now.
can you review please

**Comment by jona42-ui** [comment]
@cathay4t  this is ready for review
thanks

**Comment by ffmancera** [comment]
Hi, could you please adjust the commit name? Avoid the usage of uppercase. Just make it a normal sentence.

**Comment by jona42-ui** [comment]
> Hi, could you please adjust the commit name? Avoid the usage of uppercase. Just make it a normal sentence.

oops! correcting it now, it seems that's what the contrib docs says(if am not mistaken)

**Review by ffmancera — APPROVED** [review]

**Pull request merged**